py/desispec/brick.py

```diff
@@ -75,28 +75,30 @@ def __init__(self, bricksize=0.5):
 
     def brickname(self, ra, dec):
         """Return string name of brick that contains (ra, dec) [degrees]
-        
+
         Args:
             ra (float) : Right Ascension in degrees
             dec (float) : Declination in degrees
-            
+
         Returns:
             brick name string
         """
         inra, indec = ra, dec
         dec = np.atleast_1d(dec)
         ra = np.atleast_1d(ra)
         irow = ((dec+90.0+self._bricksize/2)/self._bricksize).astype(int)
-        names = list()
-        for i in range(len(ra)):
-            ncol = self._ncol_per_row[irow[i]]
-            j = int(ra[i]/360 * ncol)
-            names.append(self._brickname[irow[i]][j])
+
+        ncol = self._ncol_per_row[irow]
+        jj = (ra/360.0 * ncol).astype(int)
+        names = np.empty(len(ra), dtype='U8')
+        for thisrow in set(irow):
+            these = np.where(thisrow == irow)[0]
+            names[these] = np.array(self._brickname[thisrow])[jj[these]]
 
         if np.isscalar(inra):
             return names[0]
         else:
-            return np.array(names)
+            return names
```

Comment:
Commentary: In an earlier version of the code, `names` started out as a list that was appended and then was converted to an array here at the end.  In other cases where you are dealing with user input that might be a list or an array, a good trick to know is `np.asarray`: if the input was already an array it does a fast pass-through (faster than `np.array` which creates a new array), otherwise it does the slower operation of converting it to an array.  I don't think that applies in this particular case.